### DISCUSSION TITLE
Zigbee friendlyname

### DISCUSSION THREAD

**Discussion by CBSalem** [arendst/Tasmota#17956] [Q&A] [answered]
Hi guys,

is there anything like %zbdevice% for "friendlyname" of zigbee devices set through zbname, I went through all the doc, but I might have missed something. Thanks

**Comment by sfromis** [comment, marked as answer]
Such depends on where you want to use it, and how. As a Zigbee coordinator with Tasmota can support many end devices, something like `%zbdevice%` could not work. When you use `ZbName`, the name is generally included in the payloads generated from the device, which should be the primary place to find it when having activity for a particular device. Does this not cover what you want?

If you need a lookup table for `ZbName` values, the command `ZbInfo` provides a listing of devices, including relevant identifiers.

**Reply by CBSalem** [reply]
I am building a payload that I publish from ON ZBRECEIVED#?#BatteryPercentage but I need "friendlyname" and not zbdevice, and I can't get it with,  DO BACKLOG  VAR1 zbname %zbdevice%; PUBLISH topic/%VAR1% {"BatteryPercentage":"%VALUE%"} ENDON, I  get instead of friendly name just "zbname 0xFF75" only zbdevice was "expanded". Thanks for any tip

**Reply by sfromis** [reply]
What you can do in that case is to have two rules.
`on ZbReceived#?#Name do var1 %value% endon`
`on ZbReceived#?#BatteryPercentage do Publish topic/%var1% {"BatteryPercentage":"%value%"} endon`
This method also allows picking up other fields in other vars.

Of course, you do **not** need to do this at all, as you can just have your backend subscribe to the original messages, and pick fields from that.

**Reply by CBSalem** [reply]
Thanks, your method worked perfectly. 
But Backlog wasn't meant to be able to pass 30 commands at once between ON .... ENDON.

**Reply by sfromis** [reply]
Yes, Backlog is used to have a list of commands, **but** as %varx% are expanded *before* the rule starts, you cannot first set and then use the same variable within one rule. Therefore two.

**Comment by s-hadinger** [comment]
There isn't but it didn't seem necessary since all commands still accept %zbdevice% as a valid parameter.
